thin-replica-server/test/thin_replica_server_test.cpp

```diff
@@ -142,7 +142,7 @@ class FakeStorage : public concord::kvbc::IReader {
 
  private:
   BlockMap db_;
-  BlockId block_id_;
+  std::atomic<BlockId> block_id_;
```

Comment:
The same question as above.

Reply:
Data race was taken care, however race condition was missed.
Resolved here https://github.com/vmware/concord-bft/pull/1750/commits/92761f7994510c3e65d1b779036ae1aa9b7bfa3d